src/documents/tutorial-developerIntroduction/DeveloperIntroductionToInfusionFramework-EventsAndInversionOfControl.md

```diff
@@ -0,0 +1,45 @@
+---
+title: Developer Introduction to the Infusion Framework - Events and Inversion of Control
+layout: default
+category: Tutorials
+---
+
+All Infusion components support highly flexible event-driven programming. All components have basic [lifecycle events](/infusion/development/ComponentLifecycle.md) such as creation and destruction, and others may be [declared](/infusion/development/InfusionEventSystem.md#declaring-an-event-on-a-component), [fired](/infusion/development/InfusionEventSystem.md#using-events-and-listeners-procedurally) and [listened](/infusion/development/InfusionEventSystem.md#registering-a-listener-to-an-event) for by the originating component or another component.
+
+Infusion also makes extensive use of the programming concept of [Inversion of Control](/infusion/development/FrameworkConcepts.md#ioc) (IoC), a technique for organizing component dependencies and references in a distributed, flexible manner. IoC is used in many ways throughout Infusion, but an important initial concept to grasp is the use of [IoC References](/infusion/development/IoCReferences.md) when configuring components, and the concept of `{that}`.
+
+IoC references allow us to refer to other parts of a component object (or another component entirely) in a declarative, context-specific manner, with `{that}` standing in for the current component configuration.
+
+We can use a listener definition, the `onCreate` lifecycle event and IoC `{that}` to have a component say hello when it's ready, rather than needing to be manually called:
+
+<div class="infusion-docs-note">You can check out the [Live Example](http://codepen.io/waharnum/pen/LxbPQZ?editors=1111) of the code below on [CodePen](http://codepen.io/)</div>
+
+```
+fluid.defaults("fluid.helloWorld", {
+    gradeNames: ["fluid.component"],
+    listeners: {
+        // Configures the component to call its own 'sayHello' function
+        // when it's created
+        //
+        // On the left side, "onCreate.sayHello":
+        // 1) Configures the listener to listen for
+        // the onCreate lifecycle event
+        // 2) Adds an a namespace of 'sayHello' to
+        // allow multiple listeners to be attached to the
+        // same event without collision
+        //
+        // On the right side, this configures the listener
+        // to use the sayHello invoker 
+        "onCreate.sayHello": "{that}.sayHello"
+    },
+    invokers: {
+        sayHello: {
+            "this": "console",
```

Comment:
A general comment, which I hope isn't out of place or annoying:

In Flocking, years ago I stupidly made an example that uses the namespace <code>my.project</code> or something like that. Even today, I regularly see code in the wild where the developer didn't even bother to customize the namespace for their project. Most even reproduce the comment immediately above the <code>fluid.registerNamespace</code> call, which said "// Change this to something more appropriate for you project!" My guess is that 75% of the deployed Flocking applications out there are members of the <code>my.project</code> namespace. 

I think it's important to always design example code for the strange reality that there is no such thing as example code. As soon as it is published, examples will become "production code" for someone, and most likely for a lot of people.